tests/integration/logical/test_snapshot.py

```diff
@@ -0,0 +1,131 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Test 'snapshot'.
+"""
+
+import time
+import unittest
+
+from stratis_cli._errors import StratisCliDbusLookupError
+
+from .._misc import _device_list
+from .._misc import RUNNER
+from .._misc import Service
+
+
+_DEVICE_STRATEGY = _device_list(1)
+
+
+class SnapshotTestCase(unittest.TestCase):
+    """
+    Test creating a snapshot of a filesystem in a pool.  In this case
+    the snapshot should be created and no error rasied.
+    """
+    _MENU = ['--propagate', 'filesystem', 'snapshot']
+    _POOLNAME = 'deadpool'
+    _SNAPNAME = 'snapfs'
+    _FSNAME = 'fs'
+
+    def setUp(self):
+        """
+        Start the stratisd daemon with the simulator.
+        """
+        self._service = Service()
+        self._service.setUp()
+        time.sleep(1)
+        command_line = \
+           ['pool', 'create'] + [self._POOLNAME] + \
+           _DEVICE_STRATEGY.example()
+        RUNNER(command_line)
+        command_line = ['filesystem', 'create', self._POOLNAME, self._FSNAME]
+        RUNNER(command_line)
+
+    def tearDown(self):
+        """
+        Stop the stratisd simulator and daemon.
+        """
+        self._service.tearDown()
+
+    def testSnapshot(self):
+        """
+        Creation of the snapshot should succeed since origin pool/filesytem is available.
+        """
+        command_line = self._MENU + [self._POOLNAME, self._FSNAME, self._SNAPNAME]
+        RUNNER(command_line)
+
+class Snapshot1TestCase(unittest.TestCase):
+    """
+    Test creating a snapshot w/out a pool.
+    """
+    _MENU = ['--propagate', 'filesystem', 'snapshot']
+    _POOLNAME = 'nopool'
+    _SNAPNAME = 'snapfs'
+    _FSNAME = 'fs'
+
+    def setUp(self):
+        """
+        Start the stratisd daemon with the simulator.
+        """
+        self._service = Service()
+        self._service.setUp()
+        time.sleep(1)
+
+    def tearDown(self):
+        """
+        Stop the stratisd simulator and daemon.
+        """
+        self._service.tearDown()
+
+    def testCreation(self):
+        """
+        Creation of the snapshot must fail since specified pool does not exist.
+        """
+        command_line = self._MENU + [self._POOLNAME, self._FSNAME, self._SNAPNAME]
+        with self.assertRaises(StratisCliDbusLookupError):
+            RUNNER(command_line)
+
+class Snapshot2TestCase(unittest.TestCase):
+    """
+    Test creating a snapshot w/out a filesystem.
+    """
+    _MENU = ['--propagate', 'filesystem', 'snapshot']
+    _POOLNAME = 'pool'
+    _FSNAME = 'fs'
+    _SNAPNAME = 'snapfs'
+
+    def setUp(self):
+        """
+        Start the stratisd daemon with the simulator.
+        """
+        self._service = Service()
+        self._service.setUp()
+        time.sleep(1)
+        command_line = ['pool', 'create'] + [self._POOLNAME] + _DEVICE_STRATEGY.example()
```

Comment:
I think this must be able to be combined into:

```['pool',  'create', self._POOLNAME, _DEVICE_STRATEGY.example()]``` or something very close.

Reply:
ok